tests/OK/inductive_type.lp

```diff
@@ -0,0 +1,349 @@
+/////////////////////////////
+// Configuration
+/////////////////////////////
+
+constant symbol Prop : TYPE      // Type of propositions
+set declared "π"
+injective symbol π : Prop → TYPE // Interpretation of propositions in TYPE
+
+set builtin "Prop"     ≔ Prop
+set builtin "P"     ≔ π
+
+constant symbol Set : TYPE       // Type of set codes
+set declared "τ"
+injective symbol τ : Set → TYPE  // Interpretation of set codes in TYPE
+
+/////////////////////////////
+// Equality
+/////////////////////////////
+constant symbol eq {a} : τ a → τ a → Prop
+set infix 8 "=" ≔ eq
+constant symbol eq_refl {a} (x : τ a) : π (x = x)
+constant symbol eq_ind {a} (x y : τ a) : π (x = y) → Πp, π (p y) → π (p x)
+
+// Set builtins for the rewrite tactic
+set builtin "T"     ≔ τ
+set builtin "eq"    ≔ eq
+set builtin "refl"  ≔ eq_refl
+set builtin "eqind" ≔ eq_ind
+
+//////////////////////
+// Boolean type
+//////////////////////
+
+constant symbol bool : Set       // Type of booleans
+set declared "𝔹"
+definition 𝔹 ≔ τ bool
+constant symbol true  : 𝔹
+constant symbol false : 𝔹
+
+/////////////////////////////
+// Some examples of inductive types
+/////////////////////////////
+set debug +h
+
+set declared "ℕ"
+inductive ℕ : TYPE ≔
+ | cons1 : ℕ
+ | cons2 : ℕ → ℕ
+
+//set declared "ind_ℕ"
+//symbol ind_ℕ : ℕ
+type ind_ℕ
```

Comment:
Test files should not include commands like `set debug` or `type`. Use `assert` instead.

Reply:
Yes, you right but there is always a problem here.